src/deprecated.jl

```diff
@@ -1,147 +1,10 @@
-    elseif cols isa Union{AbstractString, Symbol}
-        Base.depwarn("`categorical!(df, cols)` is deprecated. " *
-                     "Use `transform!(df, cols => $categoricalstr, renamecols=false)` instead.",
-                     :categorical!)
-        return transform!(df, cols => (x -> categorical(x, compress=compress)), renamecols=false)
-    else
-        Base.depwarn("`categorical!(df, cols)` is deprecated. " *
-                     "Use `transform!(df, names(df, cols) .=> $categoricalstr, renamecols=false)` instead.",
-                     :categorical!)
-        return transform!(df, names(df, cols) .=> (x -> categorical(x, compress=compress)), renamecols=false)
-    end
-end
-
-function categorical!(df::DataFrame, cols::Union{Type, Nothing}=nothing;
-                      compress::Bool=false)
-    if compress === nothing
-        compress = false
-        categoricalstr = "categorical"
-    else
-        categoricalstr = "(x -> categorical(x, compress=$compress))"
-    end
-    if cols === nothing
-        cols = Union{AbstractString, Missing}
-        Base.depwarn("`categorical!(df)` is deprecated. " *
-                     "Use `transform!(df, names(df, $cols) .=> $categoricalstr, renamecols=false)` instead.",
-                     :categorical!)
-    else
-        Base.depwarn("`categorical!(df, T)` is deprecated. " *
-                     "Use `transform!(df, names(df, T) .=> $categoricalstr, renamecols=false)` instead.",
-                     :categorical!)
-    end
-    return transform!(df, names(df, cols) .=> (x -> categorical(x, compress=compress)), renamecols=false)
-end
-
-@deprecate DataFrame(pairs::NTuple{N, Pair}; makeunique::Bool=false,
-          copycols::Bool=true) where {N} DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
-@deprecate DataFrame(columns::NTuple{N, AbstractVector}, cnames::NTuple{N, Symbol}; makeunique::Bool=false,
-          copycols::Bool=true) where {N} DataFrame(collect(columns), collect(cnames);
-              makeunique=makeunique, copycols=copycols)
-@deprecate DataFrame(columns::NTuple{N, AbstractVector}, cnames::NTuple{N, AbstractString}; makeunique::Bool=false,
-                     copycols::Bool=true) where {N} DataFrame(collect(columns), [Symbol(c) for c in cnames];
-                                                              makeunique=makeunique, copycols=copycols)
-@deprecate DataFrame(columns::NTuple{N, AbstractVector};
-                     copycols::Bool=true) where {N} DataFrame(collect(columns),
-                                                              Symbol.(:x, 1:length(columns)), copycols=copycols)
-
-# this deprecation is very important, becuase without it users will
-# get strange results with old code as described in https://github.com/JuliaData/Tables.jl/issues/208
-@deprecate DataFrame(columns::AbstractVector{<:AbstractVector}; makeunique::Bool=false,
-                     copycols::Bool=true) DataFrame(columns, :auto, copycols=copycols)
```

Comment:
As usual I'd be tempted to keep as many deprecations as possible for some time, turning them into errors. That shouldn't be too costly and that will help people who haven't noticed the deprecations introduced in 0.22. That's likely to be the case for many people as you usually don't use strict version bounds when developing interactive scripts (as opposed to packages).

(Of course that doesn't apply to `categorical` since it would conflict with CategoricalArrays.)

Reply:
OK - this essentially applies to `DataFrame` constructors. I will re-introduce them.